Source/WarpX.cpp

```diff
@@ -874,10 +874,11 @@ WarpX::ReadParameters ()
         for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
             WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
                 !(
-                    ( WarpX::field_boundary_lo[idim] == FieldBoundaryType::PML &&
-                    WarpX::field_boundary_lo[idim] == FieldBoundaryType::Absorbing_SilverMueller ) ||
-                    ( WarpX::field_boundary_hi[idim] == FieldBoundaryType::PML &&
-                     WarpX::field_boundary_hi[idim] == FieldBoundaryType::Absorbing_SilverMueller )
+                    std::any_of(WarpX::field_boundary_lo.begin(), WarpX::field_boundary_lo.end(),
+                        [](const auto& cc){return cc == FieldBoundaryType::PML;})
+                    &&
+                    std::any_of(WarpX::field_boundary_lo.begin(), WarpX::field_boundary_lo.end(),
+                        [](const auto& cc){return cc == FieldBoundaryType::Absorbing_SilverMueller;})
```

Comment:
Thanks, @lucafedeli88! 

I think I still need to have a clearer picture of the original intention here.

Quoting from #2051:
> Thus, I include an abort if the user has specified mixed boundaries with silver-mueller at one boundary and a different boundary condition at another domain boundary. The capability to have mixed boundaries with silver-mueller will be added in a separate PR.

@RevathiJambunathan 
When it says "mixed boundaries", what does it mean exactly? Does it mean different boundary conditions at the lo and hi boundary, or different boundary conditions along different directions, or a combination of both?

I'm asking because I see that `WarpX::field_boundary_hi` doesn't appear in the changes here and I'm not sure if that matches the original intention of #2051.

Reply:
By "mixed-boundaries", we meant different boundary conditions in different directions.